src/librustc/ty/layout.rs

```diff
@@ -2374,7 +2448,14 @@ where
         + HasParamEnv<'tcx>,
 {
     fn of_fn_ptr(cx: &C, sig: ty::PolyFnSig<'tcx>, extra_args: &[Ty<'tcx>]) -> Self {
-        call::FnAbi::new_internal(cx, sig, extra_args, None, |ty, _| ArgAbi::new(cx.layout_of(ty)))
+        let conv = conv(cx.tcx(), sig.abi());
+
+        let can_unwind =
+            can_unwind(cx.tcx().sess.panic_strategy(), CodegenFnAttrFlags::empty(), conv);
```

Comment:
Do we ever apply `nounwind` to function pointer calls today? If not, then this should always be `true`.

Reply:
No, it looks like we do not (#64090 seems to be related). I'm backing this out and restoring the current behavior.